inc/plugins/hello.php

```diff
@@ -8,17 +8,58 @@
  *
  */
 
-// Disallow direct access to this file for security reasons
-if(!defined("IN_MYBB"))
+// Make sure we can't access this file directly from the browser.
+if(!defined('IN_MYBB'))
 {
-	die("Direct initialization of this file is not allowed.<br /><br />Please make sure IN_MYBB is defined.");
+	die('This file cannot be accessed directly.');
 }
+	
+// cache templates - this is important when it comes to performance
+// THIS_SCRIPT is defined by some of the MyBB scripts, including index.php
+if(defined('THIS_SCRIPT'))
+{
+    global $templatelist;
+
+    if(isset($templatelist))
+    {
+        $templatelist .= ',';
```

Comment:
I don't think it breaks anything, it only tries to select a template with empty title, which is already done here anyways: https://github.com/mybb/mybb/blob/feature/inc/class_templates.php#L49

But I agree that your code is more preferred.